superset/connectors/druid/models.py

```diff
@@ -799,11 +777,54 @@ def recursive_get_fields(_conf):
                     post_aggs[metric_name] = HyperUniqueCardinality(
                         mconf.get('name')
                     )
-                else:
+                elif mconf.get('type') == 'arithmetic':
                     post_aggs[metric_name] = Postaggregator(
                         mconf.get('fn', "/"),
                         mconf.get('fields', []),
                         mconf.get('name', ''))
+                else:
+                    post_aggs[metric_name] = Postaggregator(
+                        None,
```

Comment:
fixed by making a CustomPostAggregator class that encompasses this logic. Should make more sense that way.